### PR TITLE
[Documentation] Refactor: Update tsdoc references

### PR DESCRIPTION
**Pull request by joaquim-verges**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on enhancing the `NFT` component and its related functionalities in the `thirdweb` library, introducing new props for better loading and fallback handling, and refining the overall structure and documentation of various components.

### Detailed summary
- Deleted several test files related to `NFT` and wallet integration.
- Updated `NFT` component structure to use `NFTProvider` for context management.
- Introduced `loadingComponent` and `fallbackComponent` props in `NFTMedia`, `NFTName`, and `NFTDescription`.
- Modified documentation for components to reflect changes and improve clarity.
- Changed function annotations to improve consistency across the codebase.
- Revised sidebar links and organization in various documentation files.
- Enhanced error handling in NFT fetching logic.
- Updated example usages in documentation to reflect new component structure.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by joaquim-verges** [comment]
* **#5375** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5375?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5374** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5374?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joaquim-verges and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gregfromstl — APPROVED** [review]

**Comment by gregfromstl** [comment]
### Merge activity

* **Nov 13, 1:53 PM EST**: The merge label 'merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=thirdweb-dev&repo=js) once it meets the requirements.

File: .changeset/tall-pots-live.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+"thirdweb": minor
+---
+
+NFT component provider changed from NFT to NFT.Provider
+
+```tsx
+<NFT.Provider contract={contract} tokenId={0n}>
+  <Suspense fallback={"Loading nft..."}>
```
Comment (jnsdls) [inline .changeset/tall-pots-live.md:9]: why can `NFT.Provider` not wrap suspense (internally) and you can pass the fallback to it? this seems pretty clunky...

Comment (gregfromstl) [reply .changeset/tall-pots-live.md:9]: I agree it feels clunky, at the same time `NFT.Provider` is currently headless/UI-free which is nice. Introducing UI as props to that component feels like the wrong direction

**Review by gregfromstl — APPROVED** [review]

File: .changeset/tall-pots-live.md (hunk 1 above)
Comment (kien-ngo) [reply .changeset/tall-pots-live.md:9]: Noted

**Review by gregfromstl — APPROVED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**